With `dbt run` the `--profiles-dir` becomes relative to `--project-dir`

### Describe the bug
The `--profiles-dir` flag becomes relative to the `--project-dir` flag when running `dbt run`. It does not for `dbt debug`.

### Steps To Reproduce
Let's say we have the `profiles.yml` in the current working directory and our dbt project in a sub-directory `jaffle-shop`. Then the following command runs without a problem (if all is set-up correctly):

```bash
dbt debug --project-dir jaffle-shop/ --profiles-dir .
```

However, with `dbt run` this does not work.

```bash
dbt run --project-dir jaffle-shop/ --profiles-dir .     # can not find profile
```

This is because for `dbt run` the profiles directory argument becomes relative to the project directory. Ergo, the following **does** work:

```bash
dbt run --project-dir jaffle-shop/ --profiles-dir ..     # works unexpectedly
```

### Expected behavior
The `--profile-dir` flag should behave the same for all `dbt` commands.

### Screenshots and log output
N/A

### System information
N/A/


**The output of `dbt --version`:**
```
installed version: 0.19.0
   latest version: 0.19.0

Up to date!

Plugins:
  - bigquery: 0.19.0
  - snowflake: 0.19.0
  - redshift: 0.19.0
  - postgres: 0.19.0
  - sqlserver: 0.19.0.1
```

**The operating system you're using:**

```
ProductName:    macOS
ProductVersion: 11.2.1
BuildVersion:   20D75
```

**The output of `python --version`:**

```
Python 3.7.2
```

### Additional context
Add any other context about the problem here.

@jtcohen6 : Could you maybe help me with the following?

If I run `make test-unit` I get an error when building the `Dockerfile.test`:

```
$ make test-unit
Unit test run starting...
Building test
Step 1/17 : FROM ubuntu:18.04
 ---> c3c304cb4f22
Step 2/17 : ENV DEBIAN_FRONTEND noninteractive
 ---> Using cache
 ---> eb84108341ee
Step 3/17 : RUN apt-get update     && apt-get dist-upgrade -y     && apt-get install -y --no-install-recommends     netcat     postgresql     curl     git     ssh     software-properties-common     make     build-essential     ca-certifi
cates     libpq-dev     libsasl2-dev     libsasl2-2     libsasl2-modules-gssapi-mit     libyaml-dev     unixodbc-dev     && add-apt-repository ppa:deadsnakes/ppa     && apt-get install -y     python     python-dev     python-pip     pyth
on3.6     python3.6-dev     python3-pip     python3.6-venv     python3.7     python3.7-dev     python3.7-venv     python3.8     python3.8-dev     python3.8-venv     python3.9     python3.9-dev     python3.9-venv     && apt-get clean     
&& rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*                                                                                                                                                                                            
 ---> Running in 22f12d9c63c0
Get:1 http://archive.ubuntu.com/ubuntu bionic InRelease [242 kB]
Get:2 http://security.ubuntu.com/ubuntu bionic-security InRelease [88.7 kB]
Get:3 http://archive.ubuntu.com/ubuntu bionic-updates InRelease [88.7 kB]
Get:4 http://archive.ubuntu.com/ubuntu bionic-backports InRelease [74.6 kB]
Get:5 http://security.ubuntu.com/ubuntu bionic-security/multiverse amd64 Packages [24.5 kB]
Err:5 http://security.ubuntu.com/ubuntu bionic-security/multiverse amd64 Packages
  File has unexpected size (13220 != 24531). Mirror sync in progress? [IP: 91.189.88.142 80]
  Hashes of expected file:
   - Filesize:24531 [weak]
   - SHA256:875198db945d73c55df8022e7b8e7418e393c28fc072cc138b7be45a51d28298
   - SHA1:bf23a75361391ae602dfd00159467a709214db2a [weak]
   - MD5Sum:796b20550a47c7daf797cfe8599c3f74 [weak]
  Release file created at: Wed, 17 Mar 2021 08:00:44 +0000
Get:6 http://security.ubuntu.com/ubuntu bionic-security/restricted amd64 Packages [333 kB]
Err:6 http://security.ubuntu.com/ubuntu bionic-security/restricted amd64 Packages
  
Get:7 http://archive.ubuntu.com/ubuntu bionic/main amd64 Packages [1344 kB]
Get:8 http://security.ubuntu.com/ubuntu bionic-security/main amd64 Packages [1995 kB]
Err:8 http://security.ubuntu.com/ubuntu bionic-security/main amd64 Packages
  
Get:7 http://archive.ubuntu.com/ubuntu bionic/main amd64 Packages [1344 kB]
Err:7 http://archive.ubuntu.com/ubuntu bionic/main amd64 Packages
  File has unexpected size (13184 != 1343916). Mirror sync in progress? [IP: 91.189.88.142 80]
  Hashes of expected file:
   - Filesize:1343916 [weak]
   - SHA256:ff7fd80e902a1acfba06e7b513711da31abe915d95b3ba39ce198e02efd209e5
   - SHA1:edd753d5bd860c126016feca4e898a736999915b [weak]
   - MD5Sum:9f666ceefac581815e5f3add8b30d3b9 [weak]
  Release file created at: Thu, 26 Apr 2018 23:37:48 +0000
Get:9 http://security.ubuntu.com/ubuntu bionic-security/universe amd64 Packages [1398 kB]
Err:9 http://security.ubuntu.com/ubuntu bionic-security/universe amd64 Packages
  
Get:10 http://archive.ubuntu.com/ubuntu bionic/multiverse amd64 Packages [186 kB]
Err:10 http://archive.ubuntu.com/ubuntu bionic/multiverse amd64 Packages
  
Get:11 http://archive.ubuntu.com/ubuntu bionic/restricted amd64 Packages [13.5 kB]
Err:11 http://archive.ubuntu.com/ubuntu bionic/restricted amd64 Packages
  
Get:12 http://archive.ubuntu.com/ubuntu bionic/universe amd64 Packages [11.3 MB]
Err:12 http://archive.ubuntu.com/ubuntu bionic/universe amd64 Packages
  
Get:13 http://archive.ubuntu.com/ubuntu bionic-updates/multiverse amd64 Packages [31.4 kB]
Get:14 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 Packages [2425 kB]
Get:15 http://archive.ubuntu.com/ubuntu bionic-updates/restricted amd64 Packages [363 kB]
Get:16 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 Packages [2165 kB]
Get:17 http://archive.ubuntu.com/ubuntu bionic-backports/main amd64 Packages [11.3 kB]
Get:18 http://archive.ubuntu.com/ubuntu bionic-backports/universe amd64 Packages [11.4 kB]
Fetched 5501 kB in 2s (2501 kB/s)
Reading package lists...
E: Failed to fetch http://security.ubuntu.com/ubuntu/dists/bionic-security/multiverse/binary-amd64/by-hash/SHA256/875198db945d73c55df8022e7b8e7418e393c28fc072cc138b7be45a51d28298  File has unexpected size (13220 != 24531). Mirror sync in
 progress? [IP: 91.189.88.142 80]                                                                                                                                                                                                           
   Hashes of expected file:
    - Filesize:24531 [weak]
    - SHA256:875198db945d73c55df8022e7b8e7418e393c28fc072cc138b7be45a51d28298
    - SHA1:bf23a75361391ae602dfd00159467a709214db2a [weak]
    - MD5Sum:796b20550a47c7daf797cfe8599c3f74 [weak]
   Release file created at: Wed, 17 Mar 2021 08:00:44 +0000
E: Failed to fetch http://security.ubuntu.com/ubuntu/dists/bionic-security/restricted/binary-amd64/by-hash/SHA256/df5e3cf297eeae595f8f0975c538104edf9897c29bd64df0177f37e58adbcfe6  
E: Failed to fetch http://security.ubuntu.com/ubuntu/dists/bionic-security/main/binary-amd64/by-hash/SHA256/802a85111726038ac4d10d924e17d73e3539e1ff06a72692d369302f27c0e98f  
E: Failed to fetch http://security.ubuntu.com/ubuntu/dists/bionic-security/universe/binary-amd64/by-hash/SHA256/ec3ff33c6b45a434d7a9883732e169c555ab9a5fbb7572599c87b2f97015c4a0  
E: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/bionic/main/binary-amd64/by-hash/SHA256/ff7fd80e902a1acfba06e7b513711da31abe915d95b3ba39ce198e02efd209e5  File has unexpected size (13184 != 1343916). Mirror sync in progress? [IP
: 91.189.88.142 80]                                                                                                                                                                                                                         
   Hashes of expected file:
    - Filesize:1343916 [weak]
    - SHA256:ff7fd80e902a1acfba06e7b513711da31abe915d95b3ba39ce198e02efd209e5
    - SHA1:edd753d5bd860c126016feca4e898a736999915b [weak]
    - MD5Sum:9f666ceefac581815e5f3add8b30d3b9 [weak]
   Release file created at: Thu, 26 Apr 2018 23:37:48 +0000
E: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/bionic/multiverse/binary-amd64/by-hash/SHA256/910cb989ed0e55b8c98589881a57730a36f074123991366c0263e61582a0e156  
E: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/bionic/restricted/binary-amd64/by-hash/SHA256/81b9542ff39f796dd83159d5ef02161232ae0b766538d8fc02a299fc1ced1f4d  
E: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/bionic/universe/binary-amd64/by-hash/SHA256/ca221e8754c933c636b6c0a344617e3444a7c8cb0982ca97725cda9b7bfe1e6a  
E: Some index files failed to download. They have been ignored, or old ones used instead.
ERROR: Service 'test' failed to build : The command '/bin/sh -c apt-get update     && apt-get dist-upgrade -y     && apt-get install -y --no-install-recommends     netcat     postgresql     curl     git     ssh     software-properties-co
mmon     make     build-essential     ca-certificates     libpq-dev     libsasl2-dev     libsasl2-2     libsasl2-modules-gssapi-mit     libyaml-dev     unixodbc-dev     && add-apt-repository ppa:deadsnakes/ppa     && apt-get install -y  
   python     python-dev     python-pip     python3.6     python3.6-dev     python3-pip     python3.6-venv     python3.7     python3.7-dev     python3.7-venv     python3.8     python3.8-dev     python3.8-venv     python3.9     python3.9-
dev     python3.9-venv     && apt-get clean     && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*' returned a non-zero code: 100                                                                                                             
        6.99 real         2.88 user         0.75 sys
make: *** [test-unit] Error 1
```

Do you have the same? (If you remove the cached docker file first.)

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -47,7 +47,6 @@ Contributors:
 - [@VasiliiSurov](https://github.com/VasiliiSurov) ([#3104](https://github.com/fishtown-analytics/dbt/pull/3104))
 - [@jmcarp](https://github.com/jmcarp) ([#3145](https://github.com/fishtown-analytics/dbt/pull/3145))
 - [@bastienboutonnet](https://github.com/bastienboutonnet) ([#3151](https://github.com/fishtown-analytics/dbt/pull/3151))
-<<<<<<< HEAD
 - [@max-sixty](https://github.com/max-sixty) ([#3156](https://github.com/fishtown-analytics/dbt/pull/3156)
 - [@prratek](https://github.com/prratek) ([#3100](https://github.com/fishtown-analytics/dbt/pull/3100))
 - [@techytushar](https://github.com/techytushar) ([#3158](https://github.com/fishtown-analytics/dbt/pull/3158))

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -54,14 +54,11 @@ Contributors:
 - [@fux](https://github.com/fuchsst) ([#3241](https://github.com/fishtown-analytics/dbt/issues/3241))
 - [@dmateusp](https://github.com/dmateusp) ([#3270](https://github.com/fishtown-analytics/dbt/pull/3270))
 - [@arzavj](https://github.com/arzavj) ([3106](https://github.com/fishtown-analytics/dbt/pull/3106))
+- [@JCZuurmond](https://github.com/JCZuurmond) ([#3133](https://github.com/fishtown-analytics/dbt/pull/3133))
 
 ## dbt 0.19.1 (March 31, 2021)
 
 ## dbt 0.19.1rc2 (March 25, 2021)
-=======
-- [@JCZuurmond](https://github.com/JCZuurmond) ([#3133](https://github.com/fishtown-analytics/dbt/pull/3133))
->>>>>>> Add JCZuurmond to change log
-
 
 ### Fixes
 - Pass service-account scopes to gcloud-based oauth ([#3040](https://github.com/fishtown-analytics/dbt/issues/3040), [#3041](https://github.com/fishtown-analytics/dbt/pull/3041))

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -47,6 +47,7 @@ Contributors:
 - [@VasiliiSurov](https://github.com/VasiliiSurov) ([#3104](https://github.com/fishtown-analytics/dbt/pull/3104))
 - [@jmcarp](https://github.com/jmcarp) ([#3145](https://github.com/fishtown-analytics/dbt/pull/3145))
 - [@bastienboutonnet](https://github.com/bastienboutonnet) ([#3151](https://github.com/fishtown-analytics/dbt/pull/3151))
+<<<<<<< HEAD
 - [@max-sixty](https://github.com/max-sixty) ([#3156](https://github.com/fishtown-analytics/dbt/pull/3156)
 - [@prratek](https://github.com/prratek) ([#3100](https://github.com/fishtown-analytics/dbt/pull/3100))
 - [@techytushar](https://github.com/techytushar) ([#3158](https://github.com/fishtown-analytics/dbt/pull/3158))
@@ -58,6 +59,9 @@ Contributors:
 ## dbt 0.19.1 (March 31, 2021)
 
 ## dbt 0.19.1rc2 (March 25, 2021)
+=======
+- [@JCZuurmond](https://github.com/JCZuurmond) ([#3133](https://github.com/fishtown-analytics/dbt/pull/3133))
+>>>>>>> Add JCZuurmond to change log
 
 
 ### Fixes

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -11,6 +11,7 @@
 - Raise a proper error message if dbt parses a macro twice due to macro duplication or misconfiguration. ([#2449](https://github.com/fishtown-analytics/dbt/issues/2449), [#3165](https://github.com/fishtown-analytics/dbt/pull/3165))
 - Fix exposures missing in graph context variable. ([#3241](https://github.com/fishtown-analytics/dbt/issues/3241))
 - Ensure that schema test macros are properly processed ([#3229](https://github.com/fishtown-analytics/dbt/issues/3229), [#3272](https://github.com/fishtown-analytics/dbt/pull/3272))
+- Use absolute path for profiles directory instead of a path relative to the project directory. Note: If a user supplies a relative path to the profiles directory, the value of `args.profiles_dir` will still be absolute. ([#3133](https://github.com/fishtown-analytics/dbt/issues/3133))
 
 ### Features
 - Support commit hashes in dbt deps package revision ([#3268](https://github.com/fishtown-analytics/dbt/issues/3268), [#3270](https://github.com/fishtown-analytics/dbt/pull/3270))

--- a/core/dbt/main.py
+++ b/core/dbt/main.py
@@ -1051,7 +1051,7 @@ def parse_args(args, cls=DBTArgumentParser):
     parsed = p.parse_args(args)
 
     if hasattr(parsed, 'profiles_dir'):
-        parsed.profiles_dir = os.path.expanduser(parsed.profiles_dir)
+        parsed.profiles_dir = os.path.abspath(parsed.profiles_dir)
 
     if getattr(parsed, 'project_dir', None) is not None:
         expanded_user = os.path.expanduser(parsed.project_dir)

--- a/test/integration/015_cli_invocation_tests/test_cli_invocation.py
+++ b/test/integration/015_cli_invocation_tests/test_cli_invocation.py
@@ -5,6 +5,7 @@ import shutil
 import pytest
 import tempfile
 import yaml
+from typing import Dict
 
 
 @contextlib.contextmanager
@@ -40,11 +41,12 @@ def temporary_working_directory() -> str:
     out : str
         The temporary working directory.
     """
-    with change_working_directory(tempfile.TemporaryDirectory()) as tmpdir:
-        yield tmpdir
+    with tempfile.TemporaryDirectory() as tmpdir:
+        with change_working_directory(tmpdir):
+            yield tmpdir
 
 
-def get_custom_profile_config() -> Dict:
+def get_custom_profiles_config(database_host, custom_schema):
     return {
         "config": {
             "send_anonymous_usage_stats": False
@@ -54,12 +56,12 @@ def get_custom_profile_config() -> Dict:
                 "default": {
                     "type": "postgres",
                     "threads": 1,
-                    "host": self.database_host,
+                    "host": database_host,
                     "port": 5432,
                     "user": "root",
                     "pass": "password",
                     "dbname": "dbt",
-                    "schema": self.custom_schema
+                    "schema": custom_schema
                 },
             },
             "target": "default",
@@ -67,21 +69,25 @@ def get_custom_profile_config() -> Dict:
     }
 
 
-def create_directory_with_custom_profiles(directory: str) -> None:
+def create_directory_with_custom_profiles(
+    directory: str,
+    profiles: Dict
+) -> None:
     """
-    Create directory with profiles.yml. The profile from
-    :func:get_custom_profile_config is used.
+    Create directory with profiles.yml.
 
     Parameters
     ----------
     directory : str
         The directory in which a profiles file is created.
+    profiles : Dict
+        The profiles to put into the profiles.yml
     """
-    if not os.path.exists(profiles_dir):
-        os.makedirs(profiles_dir)
+    if not os.path.exists(directory):
+        os.makedirs(directory)
 
-    with open(f"{profiles_dir}/profiles.yml", "w") as f:
-        yaml.safe_dump(self.custom_profile_config(), f, default_flow_style=True)
+    with open(f"{directory}/profiles.yml", "w") as f:
+        yaml.safe_dump(profiles, f, default_flow_style=True)
 
 
 class ModelCopyingIntegrationTest(DBTIntegrationTest):
@@ -147,12 +153,10 @@ class TestCLIInvocationWithProfilesDir(ModelCopyingIntegrationTest):
         self.run_sql(f"DROP SCHEMA IF EXISTS {self.custom_schema} CASCADE;")
         self.run_sql(f"CREATE SCHEMA {self.custom_schema};")
 
-        # the test framework will remove this in teardown for us.
-        if not os.path.exists('./dbt-profile'):
-            os.makedirs('./dbt-profile')
-
-        with open("./dbt-profile/profiles.yml", 'w') as f:
-            yaml.safe_dump(get_custom_profile_config(), f, default_flow_style=True)
+        profiles = get_custom_profiles_config(
+            self.database_host, self.custom_schema)
+        create_directory_with_custom_profiles(
+            "./dbt-profile", profiles)
 
         self.run_sql_file("seed_custom.sql")
 

--- a/test/integration/015_cli_invocation_tests/test_cli_invocation.py
+++ b/test/integration/015_cli_invocation_tests/test_cli_invocation.py
@@ -280,6 +280,11 @@ class TestCLIInvocationWithProfilesAndProjectDir(ModelCopyingIntegrationTest):
         self._test_postgres_sub_command_with_profiles_separate_from_project_dir("deps")
 
     @use_profile("postgres")
+    def test_postgres_run_with_profiles_separate_from_project_dir(self):
+        self._test_postgres_sub_command_with_profiles_separate_from_project_dir("deps")
+        self._test_postgres_sub_command_with_profiles_separate_from_project_dir("run")
+
+    @use_profile("postgres")
     def test_postgres_test_with_profiles_separate_from_project_dir(self):
         self._test_postgres_sub_command_with_profiles_separate_from_project_dir("deps")
         self._test_postgres_sub_command_with_profiles_separate_from_project_dir("run")
@@ -288,8 +293,3 @@ class TestCLIInvocationWithProfilesAndProjectDir(ModelCopyingIntegrationTest):
     @use_profile("postgres")
     def test_postgres_debug_with_profiles_separate_from_project_dir(self):
         self._test_postgres_sub_command_with_profiles_separate_from_project_dir("debug")
-
-    @use_profile("postgres")
-    def test_postgres_run_with_profiles_separate_from_project_dir(self):
-        self._test_postgres_sub_command_with_profiles_separate_from_project_dir("deps")
-        self._test_postgres_sub_command_with_profiles_separate_from_project_dir("run")

--- a/test/integration/015_cli_invocation_tests/test_cli_invocation.py
+++ b/test/integration/015_cli_invocation_tests/test_cli_invocation.py
@@ -261,5 +261,8 @@ class TestCLIInvocationWithProfilesAndProjectDir(ModelCopyingIntegrationTest):
             create_directory_with_custom_profiles(profiles_dir, profiles)
 
             project_dir = os.path.relpath(workdir, tmpdir)
+            if os.path.exists(f"{project_dir}/profiles.yml"):
+                os.remove(f"{project_dir}/profiles.yml")
+
             for dbt_sub_command in ["deps", "debug", "run"]:
                 self.run_dbt([dbt_sub_command, "--profiles-dir", profiles_dir, "--project-dir", project_dir])

--- a/test/integration/015_cli_invocation_tests/test_cli_invocation.py
+++ b/test/integration/015_cli_invocation_tests/test_cli_invocation.py
@@ -1,8 +1,33 @@
 from test.integration.base import DBTIntegrationTest, use_profile
+import contextlib
 import os
 import shutil
 import tempfile
 import yaml
+from pathlib import Path
+
+
+@contextlib.contextmanager
+def change_working_directory(directory: Union[str, Path]) -> Union[str, Path]:
+    """
+    Context manager for changing the working directory.
+
+    Parameters
+    ----------
+    directory : Union[str, Path]
+        The directory to which the working directory should be changed.
+
+    Yields
+    ------
+    out : Union[str, Path]
+        The new working directory.
+    """
+    current_working_directory = os.getcwd()
+    os.chdir(directory)
+    try:
+        yield directory
+    finally:
+        os.chdir(current_working_directory)
 
 
 class ModelCopyingIntegrationTest(DBTIntegrationTest):

--- a/test/integration/015_cli_invocation_tests/test_cli_invocation.py
+++ b/test/integration/015_cli_invocation_tests/test_cli_invocation.py
@@ -279,7 +279,6 @@ class TestCLIInvocationWithProfilesAndProjectDir(ModelCopyingIntegrationTest):
     def test_postgres_deps_with_profiles_separate_from_project_dir(self):
         self._test_postgres_sub_command_with_profiles_separate_from_project_dir("deps")
 
-
     @use_profile("postgres")
     def test_postgres_run_with_profiles_separate_from_project_dir(self):
         self._test_postgres_sub_command_with_profiles_separate_from_project_dir("deps")

--- a/test/integration/015_cli_invocation_tests/test_cli_invocation.py
+++ b/test/integration/015_cli_invocation_tests/test_cli_invocation.py
@@ -279,6 +279,7 @@ class TestCLIInvocationWithProfilesAndProjectDir(ModelCopyingIntegrationTest):
     def test_postgres_deps_with_profiles_separate_from_project_dir(self):
         self._test_postgres_sub_command_with_profiles_separate_from_project_dir("deps")
 
+
     @use_profile("postgres")
     def test_postgres_run_with_profiles_separate_from_project_dir(self):
         self._test_postgres_sub_command_with_profiles_separate_from_project_dir("deps")

--- a/test/integration/015_cli_invocation_tests/test_cli_invocation.py
+++ b/test/integration/015_cli_invocation_tests/test_cli_invocation.py
@@ -281,6 +281,7 @@ class TestCLIInvocationWithProfilesAndProjectDir(ModelCopyingIntegrationTest):
 
     @use_profile("postgres")
     def test_postgres_test_with_profiles_separate_from_project_dir(self):
+        self._test_postgres_sub_command_with_profiles_separate_from_project_dir("deps")
         self._test_postgres_sub_command_with_profiles_separate_from_project_dir("test")
 
     @use_profile("postgres")

--- a/test/integration/015_cli_invocation_tests/test_cli_invocation.py
+++ b/test/integration/015_cli_invocation_tests/test_cli_invocation.py
@@ -250,7 +250,7 @@ class TestCLIInvocationWithProfilesAndProjectDir(ModelCopyingIntegrationTest):
     def custom_schema(self):
         return "{}_custom".format(self.unique_schema())
 
-    @use_profile("postgres")
+    @pytest.mark.profile_postgres
     def test_postgres_toplevel_dbt_run_with_profile_dir_and_project_dir(self):
         profiles_dir = "./tmp-profile"
         workdir = os.getcwd()

--- a/test/integration/015_cli_invocation_tests/test_cli_invocation.py
+++ b/test/integration/015_cli_invocation_tests/test_cli_invocation.py
@@ -6,6 +6,7 @@ import pytest
 import tempfile
 import yaml
 from pathlib import Path
+from typing import Union
 
 
 @contextlib.contextmanager

--- a/test/integration/015_cli_invocation_tests/test_cli_invocation.py
+++ b/test/integration/015_cli_invocation_tests/test_cli_invocation.py
@@ -266,14 +266,14 @@ class TestCLIInvocationWithProfilesAndProjectDir(ModelCopyingIntegrationTest):
                 self.database_host, self.custom_schema)
             create_directory_with_custom_profiles(profiles_dir, profiles)
 
-            project_dir = os.path.relpath(workdir, tmpdir)
+            project_dir = os.path.relpath(workdir, os.getcwd())
             if os.path.exists(f"{project_dir}/profiles.yml"):
                 os.remove(f"{project_dir}/profiles.yml")
 
-            args = [
+            other_args = [
                 dbt_sub_command, "--profiles-dir", profiles_dir, "--project-dir", project_dir
             ]
-            self.run_dbt(args, profiles_dir=False)
+            self.run_dbt(other_args, profiles_dir=False)
 
     @use_profile("postgres")
     def test_postgres_deps_with_profiles_separate_from_project_dir(self):

--- a/test/integration/015_cli_invocation_tests/test_cli_invocation.py
+++ b/test/integration/015_cli_invocation_tests/test_cli_invocation.py
@@ -236,13 +236,22 @@ class TestCLIInvocationWithProjectDir(ModelCopyingIntegrationTest):
             assert not os.path.isdir(target)
 
 
-class TestCLIInvocationWithProfilesAndProjectDir(TestCLIInvocationWithProfilesDir):
+class TestCLIInvocationWithProfilesAndProjectDir(ModelCopyingIntegrationTest):
 
-    @pytest.mark.parametrize("dbt_sub_command", ["deps", "debug", "run"])
-    def test_postgres_toplevel_dbt_run_with_profile_dir_and_project_dir(
-        self,
-        dbt_sub_command: str
-    ):
+    @property
+    def schema(self):
+        return "test_cli_invocation_015"
+
+    @property
+    def models(self):
+        return "models"
+
+    @property
+    def custom_schema(self):
+        return "{}_custom".format(self.unique_schema())
+
+    @use_profile("postgres")
+    def test_postgres_toplevel_dbt_run_with_profile_dir_and_project_dir(self):
         profiles_dir = "./tmp-profile"
         workdir = os.getcwd()
         with temporary_working_directory() as tmpdir:
@@ -252,4 +261,5 @@ class TestCLIInvocationWithProfilesAndProjectDir(TestCLIInvocationWithProfilesDi
             create_directory_with_custom_profiles(profiles_dir, profiles)
 
             project_dir = os.path.relpath(workdir, tmpdir)
-            self.run_dbt([dbt_sub_command, "--profiles-dir", profiles_dir, "--project-dir", project_dir])
+            for dbt_sub_command in ["deps", "debug", "run"]:
+                self.run_dbt([dbt_sub_command, "--profiles-dir", profiles_dir, "--project-dir", project_dir])

--- a/test/integration/015_cli_invocation_tests/test_cli_invocation.py
+++ b/test/integration/015_cli_invocation_tests/test_cli_invocation.py
@@ -30,7 +30,7 @@ def change_working_directory(directory: str) -> str:
         os.chdir(current_working_directory)
 
 
-@pytest.fixture
+@contextlib.contextmanager
 def temporary_working_directory() -> str:
     """
     Create a temporary working directory.

--- a/test/integration/015_cli_invocation_tests/test_cli_invocation.py
+++ b/test/integration/015_cli_invocation_tests/test_cli_invocation.py
@@ -270,7 +270,10 @@ class TestCLIInvocationWithProfilesAndProjectDir(ModelCopyingIntegrationTest):
             if os.path.exists(f"{project_dir}/profiles.yml"):
                 os.remove(f"{project_dir}/profiles.yml")
 
-            self.run_dbt([dbt_sub_command, "--profiles-dir", profiles_dir, "--project-dir", project_dir])
+            args = [
+                dbt_sub_command, "--profiles-dir", profiles_dir, "--project-dir", project_dir
+            ]
+            self.run_dbt(args, profiles_dir=False)
 
     @use_profile("postgres")
     def test_postgres_deps_with_profiles_separate_from_project_dir(self):

--- a/test/integration/015_cli_invocation_tests/test_cli_invocation.py
+++ b/test/integration/015_cli_invocation_tests/test_cli_invocation.py
@@ -236,7 +236,6 @@ class TestCLIInvocationWithProjectDir(ModelCopyingIntegrationTest):
             assert not os.path.isdir(target)
 
 
-@pytest.mark.parametrize("dbt_sub_command", ["deps", "debug", "run"])
 class TestCLIInvocationWithProfilesAndProjectDir(ModelCopyingIntegrationTest):
 
     @property
@@ -251,11 +250,15 @@ class TestCLIInvocationWithProfilesAndProjectDir(ModelCopyingIntegrationTest):
     def custom_schema(self):
         return "{}_custom".format(self.unique_schema())
 
-    @pytest.mark.profile_postgres
-    def test_postgres_toplevel_dbt_run_with_profile_dir_and_project_dir(
+    @use_profile("postgres")
+    def test_sub_command_with_profiles_separate_from_project_dir(
         self,
-        dbt_sub_command: str
     ):
+        """
+        Test if a sub command runs well when a profiles dir is separate from a
+        project dir.
+        """
+        dbt_sub_command = "deps"
         profiles_dir = "./tmp-profile"
         workdir = os.getcwd()
         with temporary_working_directory() as tmpdir:

--- a/test/integration/015_cli_invocation_tests/test_cli_invocation.py
+++ b/test/integration/015_cli_invocation_tests/test_cli_invocation.py
@@ -257,6 +257,7 @@ class TestCLIInvocationWithProfilesAndProjectDir(ModelCopyingIntegrationTest):
         """
         Test if a sub command runs well when a profiles dir is separate from a
         project dir.
+
         """
         profiles_dir = "./tmp-profile"
         workdir = os.getcwd()

--- a/test/integration/015_cli_invocation_tests/test_cli_invocation.py
+++ b/test/integration/015_cli_invocation_tests/test_cli_invocation.py
@@ -236,6 +236,7 @@ class TestCLIInvocationWithProjectDir(ModelCopyingIntegrationTest):
             assert not os.path.isdir(target)
 
 
+@pytest.mark.parametrize("dbt_sub_command", ["deps", "debug", "run"])
 class TestCLIInvocationWithProfilesAndProjectDir(ModelCopyingIntegrationTest):
 
     @property
@@ -251,7 +252,10 @@ class TestCLIInvocationWithProfilesAndProjectDir(ModelCopyingIntegrationTest):
         return "{}_custom".format(self.unique_schema())
 
     @pytest.mark.profile_postgres
-    def test_postgres_toplevel_dbt_run_with_profile_dir_and_project_dir(self):
+    def test_postgres_toplevel_dbt_run_with_profile_dir_and_project_dir(
+        self,
+        dbt_sub_command: str
+    ):
         profiles_dir = "./tmp-profile"
         workdir = os.getcwd()
         with temporary_working_directory() as tmpdir:
@@ -264,5 +268,4 @@ class TestCLIInvocationWithProfilesAndProjectDir(ModelCopyingIntegrationTest):
             if os.path.exists(f"{project_dir}/profiles.yml"):
                 os.remove(f"{project_dir}/profiles.yml")
 
-            for dbt_sub_command in ["deps", "debug", "run"]:
-                self.run_dbt([dbt_sub_command, "--profiles-dir", profiles_dir, "--project-dir", project_dir])
+            self.run_dbt([dbt_sub_command, "--profiles-dir", profiles_dir, "--project-dir", project_dir])

--- a/test/integration/015_cli_invocation_tests/test_cli_invocation.py
+++ b/test/integration/015_cli_invocation_tests/test_cli_invocation.py
@@ -282,6 +282,7 @@ class TestCLIInvocationWithProfilesAndProjectDir(ModelCopyingIntegrationTest):
     @use_profile("postgres")
     def test_postgres_test_with_profiles_separate_from_project_dir(self):
         self._test_postgres_sub_command_with_profiles_separate_from_project_dir("deps")
+        self._test_postgres_sub_command_with_profiles_separate_from_project_dir("run")
         self._test_postgres_sub_command_with_profiles_separate_from_project_dir("test")
 
     @use_profile("postgres")

--- a/test/integration/015_cli_invocation_tests/test_cli_invocation.py
+++ b/test/integration/015_cli_invocation_tests/test_cli_invocation.py
@@ -2,6 +2,7 @@ from test.integration.base import DBTIntegrationTest, use_profile
 import contextlib
 import os
 import shutil
+import pytest
 import tempfile
 import yaml
 from pathlib import Path
@@ -28,6 +29,20 @@ def change_working_directory(directory: Union[str, Path]) -> Union[str, Path]:
         yield directory
     finally:
         os.chdir(current_working_directory)
+
+
+@pytest.fixture
+def temporary_working_directory() -> str:
+    """
+    Create a temporary working directory.
+
+    Returns
+    -------
+    out : str
+        The temporary working directory.
+    """
+    with change_working_directory(tempfile.TemporaryDirectory()) as tmpdir:
+        yield tmpdir
 
 
 class ModelCopyingIntegrationTest(DBTIntegrationTest):

--- a/test/integration/015_cli_invocation_tests/test_cli_invocation.py
+++ b/test/integration/015_cli_invocation_tests/test_cli_invocation.py
@@ -69,6 +69,23 @@ def get_custom_profile_config() -> Dict:
     }
 
 
+def create_directory_with_custom_profiles(directory: str) -> None:
+    """
+    Create directory with profiles.yml. The profile from
+    :func:get_custom_profile_config is used.
+
+    Parameters
+    ----------
+    directory : str
+        The directory in which a profiles file is created.
+    """
+    if not os.path.exists(profiles_dir):
+        os.makedirs(profiles_dir)
+
+    with open(f"{profiles_dir}/profiles.yml", "w") as f:
+        yaml.safe_dump(self.custom_profile_config(), f, default_flow_style=True)
+
+
 class ModelCopyingIntegrationTest(DBTIntegrationTest):
 
     def _symlink_test_folders(self):

--- a/test/integration/015_cli_invocation_tests/test_cli_invocation.py
+++ b/test/integration/015_cli_invocation_tests/test_cli_invocation.py
@@ -286,8 +286,3 @@ class TestCLIInvocationWithProfilesAndProjectDir(ModelCopyingIntegrationTest):
     @use_profile("postgres")
     def test_postgres_debug_with_profiles_separate_from_project_dir(self):
         self._test_postgres_sub_command_with_profiles_separate_from_project_dir("debug")
-
-    @use_profile("postgres")
-    def test_postgres_run_with_profiles_separate_from_project_dir(self):
-        self._test_postgres_sub_command_with_profiles_separate_from_project_dir("deps")
-        self._test_postgres_sub_command_with_profiles_separate_from_project_dir("run")

--- a/test/integration/015_cli_invocation_tests/test_cli_invocation.py
+++ b/test/integration/015_cli_invocation_tests/test_cli_invocation.py
@@ -173,3 +173,104 @@ class TestCLIInvocationWithProjectDir(ModelCopyingIntegrationTest):
         # In case of 'dbt clean' also test that the clean-targets directories were deleted.
         for target in self.config.clean_targets:
             assert not os.path.isdir(target)
+
+
+class TestCLIInvocationWithProfilesAndProjectDir(ModelCopyingIntegrationTest):
+
+    def setUp(self):
+        super().setUp()
+
+        self.run_sql(f"DROP SCHEMA IF EXISTS {self.custom_schema} CASCADE;")
+        self.run_sql(f"CREATE SCHEMA {self.custom_schema};")
+
+        # the test framework will remove this in teardown for us.
+        if not os.path.exists('./dbt-profile'):
+            os.makedirs('./dbt-profile')
+
+        with open("./dbt-profile/profiles.yml", 'w') as f:
+            yaml.safe_dump(self.custom_profile_config(), f, default_flow_style=True)
+
+        self.run_sql_file("seed_custom.sql")
+
+    def tearDown(self):
+        self.run_sql(f"DROP SCHEMA IF EXISTS {self.custom_schema} CASCADE;")
+        super().tearDown()
+
+    def custom_profile_config(self):
+        return {
+            'config': {
+                'send_anonymous_usage_stats': False
+            },
+            'test': {
+                'outputs': {
+                    'default': {
+                        'type': 'postgres',
+                        'threads': 1,
+                        'host': self.database_host,
+                        'port': 5432,
+                        'user': 'root',
+                        'pass': 'password',
+                        'dbname': 'dbt',
+                        'schema': self.custom_schema
+                    },
+                },
+                'target': 'default',
+            }
+        }
+
+    @property
+    def schema(self):
+        return "test_cli_invocation_015"
+
+    @property
+    def custom_schema(self):
+        return "{}_custom".format(self.unique_schema())
+
+    @property
+    def models(self):
+        return "models"
+
+    @use_profile('postgres')
+    def test_postgres_dbt_commands_with_relative_dir_as_project_dir(self):
+        workdir = os.getcwd()
+        with tempfile.TemporaryDirectory() as profiles_dir:
+            profiles_dir_relative = os.path.relpath(workdir, profiles_dir)
+            shutil.move(
+                workdir + '/dbt-profile/profiles.yml',
+                profiles_dir + '/profiles.yml'
+            )
+            with tempfile.TemporaryDirectory() as project_dir:
+                os.chdir(project_dir)
+                project_dir_relative = os.path.relpath(workdir, project_dir)
+                self._run_postgres_toplevel_dbt_run_with_profile_and_project_dir_arg(
+                    project_dir_relative,
+                    profiles_dir_relative,
+                )
+                os.chdir(workdir)
+            shutil.move(
+                workdir + '/dbt-profile/profiles.yml',
+                profiles_dir + '/profiles.yml'
+            )
+
+    def _run_postgres_toplevel_dbt_run_with_profile_and_project_dir_arg(
+            self,
+            project_dir,
+            profiles_dir
+    ):
+        self.run_dbt(['deps', '--project-dir', project_dir])
+        results = self.run_dbt(
+            ['run', '--project-dir', project_dir, '--profiles-dir', profiles_dir],
+            profiles_dir=False
+        )
+        self.assertEqual(len(results), 1)
+
+        actual = self.run_sql("select id from {}.model".format(self.custom_schema), fetch='one')
+
+        expected = (1, )
+        self.assertEqual(actual, expected)
+
+        res = self.run_dbt(['test', '--profiles-dir', 'dbt-profile'], profiles_dir=False)
+
+        # make sure the test runs against `custom_schema`
+        for test_result in res:
+            self.assertTrue(self.custom_schema, test_result.node.compiled_sql)

--- a/test/integration/015_cli_invocation_tests/test_cli_invocation.py
+++ b/test/integration/015_cli_invocation_tests/test_cli_invocation.py
@@ -250,15 +250,14 @@ class TestCLIInvocationWithProfilesAndProjectDir(ModelCopyingIntegrationTest):
     def custom_schema(self):
         return "{}_custom".format(self.unique_schema())
 
-    @use_profile("postgres")
-    def test_sub_command_with_profiles_separate_from_project_dir(
+    def _test_postgres_sub_command_with_profiles_separate_from_project_dir(
         self,
+        dbt_sub_command: str
     ):
         """
         Test if a sub command runs well when a profiles dir is separate from a
         project dir.
         """
-        dbt_sub_command = "deps"
         profiles_dir = "./tmp-profile"
         workdir = os.getcwd()
         with temporary_working_directory() as tmpdir:
@@ -272,3 +271,15 @@ class TestCLIInvocationWithProfilesAndProjectDir(ModelCopyingIntegrationTest):
                 os.remove(f"{project_dir}/profiles.yml")
 
             self.run_dbt([dbt_sub_command, "--profiles-dir", profiles_dir, "--project-dir", project_dir])
+
+    @use_profile("postgres")
+    def test_postgres_deps_with_profiles_separate_from_project_dir(self):
+        self._test_postgres_sub_command_with_profiles_separate_from_project_dir("deps")
+
+    @use_profile("postgres")
+    def test_postgres_test_with_profiles_separate_from_project_dir(self):
+        self._test_postgres_sub_command_with_profiles_separate_from_project_dir("test")
+
+    @use_profile("postgres")
+    def test_postgres_debug_with_profiles_separate_from_project_dir(self):
+        self._test_postgres_sub_command_with_profiles_separate_from_project_dir("debug")

--- a/test/integration/015_cli_invocation_tests/test_cli_invocation.py
+++ b/test/integration/015_cli_invocation_tests/test_cli_invocation.py
@@ -286,3 +286,8 @@ class TestCLIInvocationWithProfilesAndProjectDir(ModelCopyingIntegrationTest):
     @use_profile("postgres")
     def test_postgres_debug_with_profiles_separate_from_project_dir(self):
         self._test_postgres_sub_command_with_profiles_separate_from_project_dir("debug")
+
+    @use_profile("postgres")
+    def test_postgres_run_with_profiles_separate_from_project_dir(self):
+        self._test_postgres_sub_command_with_profiles_separate_from_project_dir("deps")
+        self._test_postgres_sub_command_with_profiles_separate_from_project_dir("run")

--- a/test/integration/015_cli_invocation_tests/test_cli_invocation.py
+++ b/test/integration/015_cli_invocation_tests/test_cli_invocation.py
@@ -5,23 +5,21 @@ import shutil
 import pytest
 import tempfile
 import yaml
-from pathlib import Path
-from typing import Union
 
 
 @contextlib.contextmanager
-def change_working_directory(directory: Union[str, Path]) -> Union[str, Path]:
+def change_working_directory(directory: str) -> str:
     """
     Context manager for changing the working directory.
 
     Parameters
     ----------
-    directory : Union[str, Path]
+    directory : str
         The directory to which the working directory should be changed.
 
     Yields
     ------
-    out : Union[str, Path]
+    out : str
         The new working directory.
     """
     current_working_directory = os.getcwd()

--- a/test/integration/015_cli_invocation_tests/test_cli_invocation.py
+++ b/test/integration/015_cli_invocation_tests/test_cli_invocation.py
@@ -238,7 +238,7 @@ class TestCLIInvocationWithProjectDir(ModelCopyingIntegrationTest):
 
 class TestCLIInvocationWithProfilesAndProjectDir(TestCLIInvocationWithProfilesDir):
 
-    @pytest.mark.parametrize("dbt_sub_command", ["deps"])
+    @pytest.mark.parametrize("dbt_sub_command", ["deps", "debug", "run"])
     def test_postgres_toplevel_dbt_run_with_profile_dir_and_project_dir(
         self,
         dbt_sub_command: str

--- a/test/integration/015_cli_invocation_tests/test_cli_invocation.py
+++ b/test/integration/015_cli_invocation_tests/test_cli_invocation.py
@@ -46,6 +46,29 @@ def temporary_working_directory() -> str:
         yield tmpdir
 
 
+def get_custom_profile_config() -> Dict:
+    return {
+        "config": {
+            "send_anonymous_usage_stats": False
+        },
+        "test": {
+            "outputs": {
+                "default": {
+                    "type": "postgres",
+                    "threads": 1,
+                    "host": self.database_host,
+                    "port": 5432,
+                    "user": "root",
+                    "pass": "password",
+                    "dbname": "dbt",
+                    "schema": self.custom_schema
+                },
+            },
+            "target": "default",
+        }
+    }
+
+
 class ModelCopyingIntegrationTest(DBTIntegrationTest):
 
     def _symlink_test_folders(self):
@@ -114,35 +137,13 @@ class TestCLIInvocationWithProfilesDir(ModelCopyingIntegrationTest):
             os.makedirs('./dbt-profile')
 
         with open("./dbt-profile/profiles.yml", 'w') as f:
-            yaml.safe_dump(self.custom_profile_config(), f, default_flow_style=True)
+            yaml.safe_dump(get_custom_profile_config(), f, default_flow_style=True)
 
         self.run_sql_file("seed_custom.sql")
 
     def tearDown(self):
         self.run_sql(f"DROP SCHEMA IF EXISTS {self.custom_schema} CASCADE;")
         super().tearDown()
-
-    def custom_profile_config(self):
-        return {
-            'config': {
-                'send_anonymous_usage_stats': False
-            },
-            'test': {
-                'outputs': {
-                    'default': {
-                        'type': 'postgres',
-                        'threads': 1,
-                        'host': self.database_host,
-                        'port': 5432,
-                        'user': 'root',
-                        'pass': 'password',
-                        'dbname': 'dbt',
-                        'schema': self.custom_schema
-                    },
-                },
-                'target': 'default',
-            }
-        }
 
     @property
     def schema(self):

--- a/test/integration/015_cli_invocation_tests/test_cli_invocation.py
+++ b/test/integration/015_cli_invocation_tests/test_cli_invocation.py
@@ -238,7 +238,11 @@ class TestCLIInvocationWithProjectDir(ModelCopyingIntegrationTest):
 
 class TestCLIInvocationWithProfilesAndProjectDir(TestCLIInvocationWithProfilesDir):
 
-    def test_postgres_toplevel_dbt_run_with_profile_dir_and_project_dir(self):
+    @pytest.mark.parametrize("dbt_sub_command", ["deps"])
+    def test_postgres_toplevel_dbt_run_with_profile_dir_and_project_dir(
+        self,
+        dbt_sub_command: str
+    ):
         profiles_dir = "./tmp-profile"
         workdir = os.getcwd()
         with temporary_working_directory() as tmpdir:
@@ -248,4 +252,4 @@ class TestCLIInvocationWithProfilesAndProjectDir(TestCLIInvocationWithProfilesDi
             create_directory_with_custom_profiles(profiles_dir, profiles)
 
             project_dir = os.path.relpath(workdir, tmpdir)
-            self.run_dbt(["deps", "--profiles-dir", profiles_dir, "--project-dir", project_dir])
+            self.run_dbt([dbt_sub_command, "--profiles-dir", profiles_dir, "--project-dir", project_dir])

--- a/test/integration/015_cli_invocation_tests/test_cli_invocation.py
+++ b/test/integration/015_cli_invocation_tests/test_cli_invocation.py
@@ -236,102 +236,16 @@ class TestCLIInvocationWithProjectDir(ModelCopyingIntegrationTest):
             assert not os.path.isdir(target)
 
 
-class TestCLIInvocationWithProfilesAndProjectDir(ModelCopyingIntegrationTest):
+class TestCLIInvocationWithProfilesAndProjectDir(TestCLIInvocationWithProfilesDir):
 
-    def setUp(self):
-        super().setUp()
-
-        self.run_sql(f"DROP SCHEMA IF EXISTS {self.custom_schema} CASCADE;")
-        self.run_sql(f"CREATE SCHEMA {self.custom_schema};")
-
-        # the test framework will remove this in teardown for us.
-        if not os.path.exists('./dbt-profile'):
-            os.makedirs('./dbt-profile')
-
-        with open("./dbt-profile/profiles.yml", 'w') as f:
-            yaml.safe_dump(self.custom_profile_config(), f, default_flow_style=True)
-
-        self.run_sql_file("seed_custom.sql")
-
-    def tearDown(self):
-        self.run_sql(f"DROP SCHEMA IF EXISTS {self.custom_schema} CASCADE;")
-        super().tearDown()
-
-    def custom_profile_config(self):
-        return {
-            'config': {
-                'send_anonymous_usage_stats': False
-            },
-            'test': {
-                'outputs': {
-                    'default': {
-                        'type': 'postgres',
-                        'threads': 1,
-                        'host': self.database_host,
-                        'port': 5432,
-                        'user': 'root',
-                        'pass': 'password',
-                        'dbname': 'dbt',
-                        'schema': self.custom_schema
-                    },
-                },
-                'target': 'default',
-            }
-        }
-
-    @property
-    def schema(self):
-        return "test_cli_invocation_015"
-
-    @property
-    def custom_schema(self):
-        return "{}_custom".format(self.unique_schema())
-
-    @property
-    def models(self):
-        return "models"
-
-    @use_profile('postgres')
-    def test_postgres_dbt_commands_with_relative_dir_as_project_dir(self):
+    def test_postgres_toplevel_dbt_run_with_profile_dir_and_project_dir(self):
+        profiles_dir = "./tmp-profile"
         workdir = os.getcwd()
-        with tempfile.TemporaryDirectory() as profiles_dir:
-            profiles_dir_relative = os.path.relpath(workdir, profiles_dir)
-            shutil.move(
-                workdir + '/dbt-profile/profiles.yml',
-                profiles_dir + '/profiles.yml'
-            )
-            with tempfile.TemporaryDirectory() as project_dir:
-                os.chdir(project_dir)
-                project_dir_relative = os.path.relpath(workdir, project_dir)
-                self._run_postgres_toplevel_dbt_run_with_profile_and_project_dir_arg(
-                    project_dir_relative,
-                    profiles_dir_relative,
-                )
-                os.chdir(workdir)
-            shutil.move(
-                workdir + '/dbt-profile/profiles.yml',
-                profiles_dir + '/profiles.yml'
-            )
+        with temporary_working_directory() as tmpdir:
 
-    def _run_postgres_toplevel_dbt_run_with_profile_and_project_dir_arg(
-            self,
-            project_dir,
-            profiles_dir
-    ):
-        self.run_dbt(['deps', '--project-dir', project_dir])
-        results = self.run_dbt(
-            ['run', '--project-dir', project_dir, '--profiles-dir', profiles_dir],
-            profiles_dir=False
-        )
-        self.assertEqual(len(results), 1)
+            profiles = get_custom_profiles_config(
+                self.database_host, self.custom_schema)
+            create_directory_with_custom_profiles(profiles_dir, profiles)
 
-        actual = self.run_sql("select id from {}.model".format(self.custom_schema), fetch='one')
-
-        expected = (1, )
-        self.assertEqual(actual, expected)
-
-        res = self.run_dbt(['test', '--profiles-dir', 'dbt-profile'], profiles_dir=False)
-
-        # make sure the test runs against `custom_schema`
-        for test_result in res:
-            self.assertTrue(self.custom_schema, test_result.node.compiled_sql)
+            project_dir = os.path.relpath(workdir, tmpdir)
+            self.run_dbt(["deps", "--profiles-dir", profiles_dir, "--project-dir", project_dir])

